tests/test_transpiler_router.py

```diff
@@ -186,6 +186,73 @@ def test_routing_with_measurements():
     )
 
 
+def test_sabre_looping():
+    # Setup where the looping occurs
+    # Line connectivity, gates with gate_array, Trivial placer
+    gate_array = [(7, 2), (6, 0), (5, 6), (4, 8), (3, 5), (9, 1)]
+    loop_circ = Circuit(10)
+    for qubits in gate_array:
+        loop_circ.add(gates.CZ(*qubits))
+
+    chip = nx.Graph()
+    chip.add_nodes_from([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    chip.add_edges_from(
+        [(0, 1), (1, 2), (2, 3), (3, 4), (4, 5), (5, 6), (6, 7), (7, 8), (8, 9)]
+    )
+
+    placer = Trivial(connectivity=chip)
+    initial_layout = placer(loop_circ)
+    router_no_threshold = Sabre(
+        connectivity=chip, swap_threshold=np.inf
+    )  # Without reset
+    router_threshold = Sabre(connectivity=chip)  # With reset
+    routed_no_threshold, _ = router_no_threshold(
+        loop_circ, initial_layout=initial_layout
+    )
+    routed_threshold, _ = router_threshold(loop_circ, initial_layout=initial_layout)
+
+    def count_swaps(circuit):
```

Comment:
You can use router.added_swaps() that returns the number of swaps added during the process without defining this function